api/v1beta1/cluster_types.go

```diff
@@ -238,6 +242,66 @@ type MachineHealthCheckTopology struct {
 	MachineHealthCheckClass `json:",inline"`
 }
 
+// MachinePoolTopology specifies the different parameters for a pool of worker nodes in the topology.
+// This pool of nodes is managed by a MachinePool object whose lifecycle is managed by the Cluster controller.
+type MachinePoolTopology struct {
+	// Metadata is the metadata applied to the MachinePool.
+	// At runtime this metadata is merged with the corresponding metadata from the ClusterClass.
```

Comment:
Note for a follow-up, when this is merged/implemented we have to update the corresponding documentation in the book

Reply:
We have an umbrella issue here https://github.com/kubernetes-sigs/cluster-api/issues/5991#issue-1114311094 including an update docs section. Are we looking for something specific?